source/docs/software/actuators/programming-leds.rst

```diff
@@ -0,0 +1,147 @@
+
+.. important:: It is important to note that setting the length of the LED header is an expensive task and it's **not** recommended to run this periodically.
+
+After the length of the strip has been set, you'll have to create an ``AddressableLEDBuffer`` object that takes the number of LEDs as an input. You'll then call ``myAddressableLed.setData(myAddressableLEDBuffer)`` to set the led output data. Finally, you can call ``myAddressableLed.start`` to write the output continuously. Below is a full example of the initialization process.
+
+.. note:: C++ does not have an AddressableLEDBuffer, and instead uses an Array.
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/addressableled/Robot.java
+         :language: java
+         :lines: 8-23
+         :linenos:
+         :lineno-start: 14
+
+   .. group-tab:: C++
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
+         :language: cpp
+         :lines: 14-23
+         :linenos:
+         :lineno-start: 14
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
+         :language: cpp
+         :lines: 41-47
+         :linenos:
+         :lineno-start: 41
+
+Setting the Entire Strip to One Color
+-------------------------------------
+
+Color can be set to an individual led on the strip using two methods.``setHSV()`` which takes HSV values as an input, and ``setRGB`` which takes RGB values as an input.
+
+Using HSV Values
+^^^^^^^^^^^^^^^^
+
+HSV stands for Hue, Saturation, and Value. Hue describes the color or tint, saturation being the amount of gray, and value being the brightness. LEDs can be set with the ``setHSV`` method that takes 4 arguments: index of the LED, hue, saturation, and value. An example is shown below for setting the color of an LED strip to red.
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+
+         for (var i = 0; i < m_ledBuffer.getLength(); i++) {
+            // Sets the specified LED to the HSV values for red
+            m_ledBuffer.setHSV(i, 0, 100, 100);
+         }
+
+         m_led.setData(m_ledBuffer);
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+
+         for (int i = 0; i < kLength; i++) {
+            m_ledBuffer[i].SetHSV(0, 100, 100);
+         }
+
+         m_led.SetData(m_ledBuffer);
+
+Using RGB Values
+^^^^^^^^^^^^^^^^
+
+RGB stands for Red, Green, and Blue. This is a fairly common color model as it's quite easy to understand. LEDs can be set with the ``setRGB`` method that takes 4 arguments: index of the LED, amount of red, amount of green, amount of blue.
```

Comment:
I think we should specify what the values are out of. Some libraries use values from 0-1, but WPILib uses integers from 0-255.